python/ribasim/model.py

```diff
@@ -31,6 +31,16 @@
 )
 
 
+class Solver(BaseModel):
+    algorithms: Optional[str]
```

Comment:
Does it make sense to make all fields optional?
I feel like at least `algorithms` should be required.

Also, why `algorithms` and not `algorithm`?

Reply:
Thanks, that should be singular, fixed.

They are all optional in the sense that in the julia code we pick reasonable defaults in the Solver struct. I could copy those defaults here, but that would mean they are always written out to the TOML file if a model is generated from Python. Do we want big explicit TOMLs with everything spelled out, or do we want to omit optional keys and leave the core to fill them in?

Reply:
Fair enough.

Reply:
Why do you think algorithm should be required? I think most users don't know or don't care about this, I don't want to force them to pick one, we can pick one for them that works well in most scenarios (though perhaps not the best in all).

Reply:
Agree, we pick a default that works in most situations but allow the modeller to change  as needed

Reply:
I think the status quo is fine for me as is.

But making algorithms required wouldn't change that Solver as a whole is optional.
It would just force people who mess with solver settings to also specify the algorithm

Reply:
Right, I understand. Though only the autodiff setting directly relates to the algorithm (i.e. is a keyword of).